unixfs/mod/dagmodifier_test.go

```diff
@@ -415,164 +411,92 @@ func TestDagTruncate(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-
-	_, err = dagmod.Seek(0, os.SEEK_SET)
+	size, err := dagmod.Size()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	out, err := ioutil.ReadAll(dagmod)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if err = arrComp(out, b[:12345]); err != nil {
-		t.Fatal(err)
+	if size != 12345 {
+		t.Fatal("size was incorrect!")
 	}
-}
 
-func TestSparseWrite(t *testing.T) {
-	dserv, pins := getMockDagServ(t)
-	_, n := getNode(t, dserv, 0, pins)
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	dagmod, err := NewDagModifier(ctx, n, dserv, pins, sizeSplitterGen(512))
+	_, err = dagmod.Seek(0, os.SEEK_SET)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	buf := make([]byte, 5000)
-	u.NewTimeSeededRand().Read(buf[2500:])
-
-	wrote, err := dagmod.WriteAt(buf[2500:], 2500)
+	out, err := ioutil.ReadAll(dagmod)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	if wrote != 2500 {
-		t.Fatal("incorrect write amount")
-	}
-
-	_, err = dagmod.Seek(0, os.SEEK_SET)
-	if err != nil {
+	if err = arrComp(out, b[:12345]); err != nil {
 		t.Fatal(err)
 	}
 
-	out, err := ioutil.ReadAll(dagmod)
+	err = dagmod.Truncate(10)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	if err = arrComp(out, buf); err != nil {
-		t.Fatal(err)
-	}
-}
-
-func basicGC(t *testing.T, bs blockstore.GCBlockstore, pins pin.Pinner) {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel() // in case error occurs during operation
-	out, err := gc.GC(ctx, bs, pins)
+	size, err = dagmod.Size()
 	if err != nil {
 		t.Fatal(err)
 	}
-	for range out {
+
+	if size != 10 {
+		t.Fatal("size was incorrect!")
 	}
 }
 
-func TestCorrectPinning(t *testing.T) {
```

Comment:
since we dont do indirect pinning, the dagmodifier doesnt need to be responsible for any sort of pinning anymore, its great!